assets/javascripts/discourse/components/user-nationalflags-preferences.js.es6

```diff
@@ -0,0 +1,29 @@
+import 'discourse/models/store'
+import { default as computed, on } from 'ember-addons/ember-computed-decorators';
+import { ajax } from 'discourse/lib/ajax';
+
+export default Ember.Component.extend({
+  layoutName: 'javascripts/discourse/templates/connectors/user-custom-preferences/user-nationalflags-preferences',
+  natflaglist: [],
+
+  @on('init')
+  setup() {
+
+    ajax({
```

Comment:
@merefield This is fine, but the more natural place for this ajax call is in the ``afterModel`` hook in a user-preferences route, added via the pluginApi in an initializer, then adding the data to the model which is passed to this plugin outlet, e.g.

```
      api.modifyClass('route:preferences', {
        afterModel(model) {
          return ajax('/natflags/flags').then(result => {
             localisedFlags = natflags.flags.map (element => {
                return {  code: element.code, 
                  pic: element.pic, 
                  description: I18n.t(`flags.description.${element.code}`)
               }
             });
            model.set('localisedFlags', localisedFlags);
          })
        }
      });
```

Reply:
OK sure, done.